### PR TITLE
CORTX-28876:Config phase of RGW mini-provisioning fails to validate 'rgw backend store' value.

### PR DESCRIPTION
**Pull request by shriya-deshmukh**
…gw backend store value

Signed-off-by: Shriya Deshmukh <shriya.deshmukh@seagate.com>

# Problem Statement
- Bug id: CORTX-28876:Config phase of RGW mini-provisioning fails to validate rgw backend store value

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/const.py [hunk 1]
```diff
@@ -25,6 +25,7 @@
 RGW_CONF_TMPL = f'{RGW_INSTALL_PATH}/conf/cortx_{COMPONENT_NAME}.conf'
 # e.g RGW_CONF_TMPL will be /opt/seagate/cortx/rgw/cof/cortx_rgw.conf
 RGW_CONF_FILE = f'cortx_{COMPONENT_NAME}.conf'
+BACKEND_STORE = 'motr'
```
Comment (cdeshmukh) [inline src/const.py:28]: cant hardcode btw, use it from cortx_rgw.conf

**Review by sachitanands — APPROVED** [review]

**Review by rahul-tripathi-git — APPROVED** [review]

File: src/setup/rgw.py [hunk 2]
```diff
@@ -132,6 +136,13 @@ def config(conf: MappedConf):
     def start(conf: MappedConf, index: str):
         """Create rgw admin user and start rgw service."""
 
+        # Before creating user/starting service,Verify backend store value=motr in rgw config file.
+        config_file = Rgw._get_rgw_config_path(conf)
+        Rgw._load_rgw_config(Rgw._rgw_conf_idx, f'ini://{config_file}')
+        if not Conf.get(Rgw._rgw_conf_idx, 'client>rgw backend store') == BACKEND_STORE:
+            raise SetupError(errno.EINVAL,
+                f'In {config_file}, "client>rgw backend store" key value should be "{BACKEND_STORE}".')
```
Comment (cdeshmukh) [inline src/setup/rgw.py:144]: Say 'Supported rgw backend store are ["{BACKEND_STORE}"], currently configured one is <whatever set vaue>

Above supported store print as list so we can add to that list in future if needed

File: src/const.py (hunk 1 above)
Comment (cdeshmukh) [reply src/const.py:30]: Got it, based on this added comment https://github.com/Seagate/cortx-rgw-integration/pull/32/files#r810778639,
after addressing that am good

File: src/setup/rgw.py (hunk 2 above)
Comment (shriya-deshmukh) [reply src/setup/rgw.py]: Done.

**Review by cdeshmukh — APPROVED** [review]

**Review by cdeshmukh — APPROVED** [review]

**Pull request merged**